refs/DoD.md

```diff
@@ -0,0 +1,47 @@
+# Definition of Done
+
+This document delivers a list of criteria that has to be met before any development on a blindnet project can be considered done.
+
```

Comment:
```suggestion
We operate under both of the following principles:

- "We value working software over comprehensive documentation" from the [Agile Manifesto](https://agilemanifesto.org/)
- "if it's Not Documented, It's Not Done"

```

Reply:
IMHO, this would only add confusion, as those two principles seem to refer to 2 different kinds of documentation, for 2 different audiences.

1. As the Agile Manifesto is mostly here to avoid the traditional strict dev processes, I'd consider this quote to refer to **technical** documentation for the developers who are making the software (i.e. specs), which often lead to overcomplicated and restricting dev processes.
2. This principle (and the rest of the document) refer to user (i.e. 3rd-party developers) documentation, which is a whole other topic.

(1) should indeed be avoided as must as possible.
(2) on the other hand, is mandatory for 3rd-party developers to understand and use our software.

Do you believe we should add an infobox to clarify this point?

Reply:
I have the impression that this interpretation you gave of the agile principle is not really how I see it. It may also not be how other people see it, so I think it might turn-out counterproductive to take this position and include the infobox.

My view is exactly the opposite of yours. I read the agile principle as it is written. We value working software over comprehensive documentation. In any context, and regardless of the type of documentation. 

The principle does not say that the documentation for 3rd party developers is not mandatory. IMHO it would be a misinterpretation to read the principle as something making documentation optional.

I have the impression that the "Definition of Done" proposed here aims to enforce (among other) the writing of documentation. That is indeed what we need. It is my view that such an enforcement can only be effective if is purposeful and nuanced. It means that to be followed and adopted by our team, the definition must seek to explicitely avoid any interpretation that would put it in conflict with well-established (and important) other principles that we abide to.

How it is written, it clearly leaves room for thinking that we are not agile when we operate under this Definition of Done. Your proposal to add an alternative interpretation of the agile principle (which is clearly not generally consensual - as at least I don't see it that way) is just an illustration of the difficulty people will have to identify themselves with this definition of done as it is written.

For me, personally, there is no negotiation possible with any of the agile principles. I operate under those principles even in other (non-software) areas of my work. It is very important to me. 

Equaly important as the documentation principle is the process principle ("We value individuals and interactions over processes and tools"). We have quite some processes in this company, under our openness framework. They bring us efficiency. But we don't value them over individuals and interactions.

"Responding to change over following a plan" as well. This team has adapted, changed and delivered. It does not mean there is no plan, and that the plan is not inteneded to be followed.

Being a part of this company requires a degree of maturity sufficient to tollerate seamingly paradoxiacal realities as described above. Each of us must cultivate the quality of intelectual presence at every situation, sufficient to blend those seemingly paradoxiacal principles as see them as non-conflicting without redefining them, and thus respond to every situation appropriately.

I think this decision requires more work, and implication of the team. We operate under a "de-facto" imperative of writing documentation. We can take more time to settle on a formal definition after the engineering team has had time to contribute to it.

Reply:
A formal definition can wait indeed.

To be clear, this is what I'm talking about: https://github.com/blindnet-io/privacy-computation-engine/pull/11

Documentation which is maintained and evolves alongside the code, as part of the development of any feature or bug fix. Nothing more than that.

Integrating this in any informal definition of done would be a big plus for me, as:

1. any other approach (especially, external markdown) is a nightmare to maintain
2. such information is way harder to obtain after development
3. 3rd-party developers expect this